ros2_ws/Dockerfile

```diff
@@ -5,23 +5,35 @@ ENV DEBIAN_FRONTEND=noninteractive
 RUN apt-get update && apt-get install -y \
   autoconf \
   automake \
+  build-essential \
```

Comment:
Why do we need so many additional libraries?

Reply:
This is the recommended set of libraries for installing python from source, which includes the compiler optimisations, as well as system libraries that the python built-ins use at runtime (like libssl for example). For `build-essential` and some others I am almost certain they are already there in the base image.

We could also install from package `apt-get install python3.10` but from what I saw this anyway will download a lot of these libraries required for runtime and testing, so it doesn't save that much space or time in the grand scheme of things. Also the specific versions of python available on apt depends on the apt repository, not all are always easily available.